src/main/java/org/folio/rest/helper/FundsHelper.java

```diff
@@ -100,23 +99,37 @@ public FundsHelper(Map<String, String> okapiHeaders, Context ctx, String lang) {
       });
   }
 
-  private CompletableFuture<String> getCurrentFiscalYearId(String ledgerId) {
+  public CompletableFuture<FiscalYear> getCurrentFiscalYear(String ledgerId) {
     FiscalYearsHelper fiscalYearsHelper = new FiscalYearsHelper(httpClient, okapiHeaders, ctx, lang);
+    return getTwoFirstFiscalYears(ledgerId, fiscalYearsHelper)
+      .thenApply(twoFirstFiscalYears -> {
+        if(CollectionUtils.isNotEmpty(twoFirstFiscalYears)) {
+          if(twoFirstFiscalYears.size() > 1 && isOverlapped(twoFirstFiscalYears.get(0), twoFirstFiscalYears.get(1))) {
+            return twoFirstFiscalYears.get(1);
+          } else {
+            return twoFirstFiscalYears.get(0);
+          }
+        } else {
+          return null;
+        }
+      });
+  }
+
+  private boolean isOverlapped(FiscalYear first, FiscalYear second) {
+    return first.getPeriodEnd().after(second.getPeriodStart());
+  }
+
+  private CompletableFuture<List<FiscalYear>> getTwoFirstFiscalYears(String ledgerId, FiscalYearsHelper fiscalYearsHelper) {
```

Comment:
nitpick: it would be more readable if the method was called getFirstTwoFiscalYears